OpenRA.Mods.Common/Traits/World/Pathfinder/PathFinderCacheDecorator.cs

```diff
@@ -0,0 +1,100 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class PathFinderCacheDecorator : IPathFinder
+	{
+		readonly IPathFinder pathFinder;
+		readonly ICacheStorage<List<CPos>> cacheStorage;
+
+		public PathFinderCacheDecorator(IPathFinder pathFinder, ICacheStorage<List<CPos>> cacheStorage)
+		{
+			this.pathFinder = pathFinder;
+			this.cacheStorage = cacheStorage;
+		}
+
+		public List<CPos> FindUnitPath(CPos from, CPos target, IActor self)
+		{
+			using (new PerfSample("Pathfinder"))
+			{
+				var key = "FindUnitPath" + self.ActorID + from.X + from.Y + target.X + target.Y;
```

Comment:
We could, but as I mentioned somewhere, this cache is pretty useless and the plan is to get rid of it in favor of other types of caching. No use of wasting more time on it.